engine/include/cubos/engine/assets/assets.hpp

```diff
@@ -13,6 +13,8 @@
 
 #include <cubos/core/memory/guards.hpp>
 #include <cubos/core/memory/type_map.hpp>
+#include <cubos/core/reflection/type.hpp>
```

Comment:
Not sure but I think you only need the header below (`reflect`).

Reply:
Well it uses the `Type` class that comes from `type.hpp`

Reply:
But does it use any of its methods? `reflect.hpp` forward declares it, if you're just storing references to it its enough to include the other header.